### DISCUSSION TITLE
Lazy load or use cached computed properties

### DISCUSSION THREAD

**Discussion by gizzmojo** [joshhanley/livewire-autocomplete#25] [Help]
We're using this inside a dropdown so it's not nessesarry to make any requests until the dropdown has been opened. 

Tried ading `<x-lwa::autocomplete lazy />` but this doesn't make any difference. 

Debugging this the initial requests are coming from this line so wondering if the results can be cached/optimized?
```
results: $wire.$entangle('{{ $resultsProperty->value }}', true),
```

**Comment by joshhanley** [comment]
@gizzmojo the component isn't a Livewire component (it's a blade component), so adding `lazy` won't do anything.

I'm currently working on a new version which will work better with V3 and allow you more control to use computed properties or handle when you want the data to load.

**Reply by gizzmojo** [reply]
Great thank you can't wait to see that

**Reply by joshhanley** [reply]
@gizzmojo if you want to have a look at the new version, you can look at the `next` branch here https://github.com/joshhanley/livewire-autocomplete/tree/next

Due to a complete API change, I haven't decided yet whether to make this a new major version (v3) or release it as a seperate package which will allow people to slowly migrate to the new syntax. I personally hate nothing worse than to have to change everything when I upgrade a package.

So I need to find a way to make this API update nice and smooth and non-blocking for people.
